packages/@uppy/core/src/Restricter.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable max-classes-per-file, class-methods-use-this */
+/* global AggregateError */
+const prettierBytes = require('@transloadit/prettier-bytes')
+const match = require('mime-match')
+
+const defaultOptions = {
+  maxFileSize: null,
+  minFileSize: null,
+  maxTotalFileSize: null,
+  maxNumberOfFiles: null,
+  minNumberOfFiles: null,
+  allowedFileTypes: null,
+  requiredMetaFields: [],
+}
+
+class RestrictionError extends Error {
+  isRestriction = true
+}
+
+if (typeof AggregateError === 'undefined') {
+  // eslint-disable-next-line no-global-assign
+  // TODO: remove this "polyfill" in the next major.
+  globalThis.AggregateError = class AggregateError extends Error {
+    constructor (errors, message) {
+      super(message)
+      this.errors = errors
+    }
+  }
+}
+
+class Restricter {
```

Comment:
now that all the methods are named "validateX" should we also rename this to Validator? :)

Reply:
I think Restricter is conceptually okay because it exclusively deals with `opts.restrictions` and it throws errors, restricting an upload.